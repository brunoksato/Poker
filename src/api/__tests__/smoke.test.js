const socketCtrl = require('../socket');
const store = require('../../models/store');
const deck = require('../../utils/deck');

describe('basic test with 2 players', () => {
  let mockSocket;
  let mockCallback;
  let mockIO;
  let mockEmit;
  beforeAll(() => {
    mockEmit = jest.fn();
    mockIO = {
      to: jest.fn(() => ({ emit: mockEmit })),
      sockets: {
        on: jest.fn(),
        in: jest.fn(() => ({
          emit: mockEmit,
        })),
      },
    };
    mockSockets = [
      {
        id: 'sid_player1',
        join: jest.fn(),
      },
      {
        id: 'sid_player2',
        join: jest.fn(),
      },
    ];
    mockCallback = jest.fn();
    jest
      .spyOn(deck, 'shuffle')
      .mockImplementation(() => [
        '7d',
        '9c',
        '2h',
        'Ts',
        'Js',
        '3c',
        '4s',
        '8s',
        'Qs',
        '9s',
        '5c',
        'Ad',
        '9d',
        'Jc',
        'Kh',
        '5h',
        'Tc',
        'Qc',
        'Qd',
        'Ac',
        '5s',
        '9h',
        'As',
        '7c',
        'Ks',
        '3d',
        'Jh',
        '4d',
        'Qh',
        '8h',
        '6s',
        '2d',
        '6h',
        '6d',
        '6c',
        'Td',
        '7h',
        'Th',
        '8d',
        '3s',
        '3h',
        'Kc',
        '2s',
        '7s',
        '4c',
        'Jd',
        '8c',
        '2c',
        'Ah',
        'Kd',
        '5d',
        '4h',
      ]);

    socketCtrl.init(mockIO);
  });
  afterEach(() => {
    jest.clearAllMocks();
  });
  afterAll(() => {});

  test('register Player 1', () => {
    const socket = mockSockets[0];
    socketCtrl.handleRegister('Player 1', socket, mockCallback);
    expect(mockCallback).toHaveBeenCalledWith({
      success: true,
      screenName: 'Player 1',
      totalChips: 10000,
    });
    expect(store.getState()).toMatchSnapshot();
  });

  test('register Player 2', () => {
    const socket = mockSockets[1];
    socketCtrl.handleRegister('Player 2', socket, mockCallback);
    expect(mockCallback).toHaveBeenCalledWith({
      success: true,
      screenName: 'Player 2',
      totalChips: 10000,
    });
    expect(store.getState()).toMatchSnapshot();
  });

  test('player 1 enters room', () => {
    const socket = mockSockets[0];
    socketCtrl.handleEnterRoom(0, socket);
    expect(socket.join).toHaveBeenCalledWith(`table-0`);
    expect(store.getState()).toMatchSnapshot();
  });

  test('player 2 enters room', () => {
    const socket = mockSockets[1];
    socketCtrl.handleEnterRoom(0, socket);
    expect(socket.join).toHaveBeenCalledWith(`table-0`);
    expect(store.getState()).toMatchSnapshot();
  });

  test('player 1 takes seat on 3', () => {
    const socket = mockSockets[0];
    socketCtrl.handleSitOnTheTable(
      {
        seat: 3,
        tableId: 0,
        chips: 500,
      },
      mockCallback,
      socket
    );
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('player 2 takes seat on 5', () => {
    const socket = mockSockets[1];
    const spyRandom = jest.spyOn(Math, 'random');
    spyRandom.mockImplementation(() => 0);
    socketCtrl.handleSitOnTheTable(
      {
        seat: 5,
        tableId: 0,
        chips: 500,
      },
      mockCallback,
      socket
    );

    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player1');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('player 1 post small blind', () => {
    const socket = mockSockets[0];
    socketCtrl.handlePostBlind(null, mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player2');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('player 2 post big blind', () => {
    const socket = mockSockets[1];
    socketCtrl.handlePostBlind(null, mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player1');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('player 1 calls small blind with 5', () => {
    const socket = mockSockets[0];
    socketCtrl.handleCall(mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player2');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('player 2 (BB) checks', () => {
    const socket = mockSockets[1];
    socketCtrl.handleCheck(mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player2');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('Flop: player 2 (BB) bets 15', () => {
    const socket = mockSockets[1];
    socketCtrl.handleBet(15, mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player1');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('Flop: player 1 (BB) raises 40', () => {
    const socket = mockSockets[0];
    socketCtrl.handleBet(40, mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player2');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });

  test('Flop: player 2 (BB) calls 25', () => {
    const socket = mockSockets[1];
    socketCtrl.handleCall(mockCallback, socket);
    expect(mockCallback).toHaveBeenCalledWith({ success: true });
    expect(mockIO.sockets.in).toHaveBeenCalledWith('table-0');
    expect(mockIO.to).toHaveBeenCalledWith('sid_player2');
    expect(store.getState().tables.byId[0]).toMatchSnapshot();
  });
});
