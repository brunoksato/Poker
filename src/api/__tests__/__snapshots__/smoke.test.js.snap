// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic test with 2 players player 1 calls small blind with 5 1`] = `
Object {
  "activeSeatsCount": 2,
  "bigBlind": 10,
  "biggestBet": 10,
  "board": Array [],
  "button": 3,
  "deck": Array [
    "7d",
    "9c",
    "2h",
    "Ts",
    "Js",
    "3c",
    "4s",
    "8s",
    "Qs",
    "9s",
    "5c",
    "Ad",
    "9d",
    "Jc",
    "Kh",
    "5h",
    "Tc",
    "Qc",
    "Qd",
    "Ac",
    "5s",
    "9h",
    "As",
    "7c",
    "Ks",
    "3d",
    "Jh",
    "4d",
    "Qh",
    "8h",
    "6s",
    "2d",
    "6h",
    "6d",
    "6c",
    "Td",
    "7h",
    "Th",
    "8d",
    "3s",
    "3h",
    "Kc",
    "2s",
    "7s",
    "4c",
    "Jd",
    "8c",
    "2c",
    "Ah",
    "Kd",
    "5d",
    "4h",
  ],
  "gameOn": true,
  "id": 0,
  "lastPlayerToAct": 5,
  "maxActionTimeout": 60000,
  "maxBuyIn": 1000,
  "minActionTimeout": 15000,
  "minBuyIn": 500,
  "name": "Winna winna chicken dinna: 5/10 aar",
  "phase": "preFlop",
  "privateTable": false,
  "seats": Array [
    null,
    null,
    null,
    Object {
      "bet": 10,
      "cards": Array [],
      "chipsInPlay": 490,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player1",
      "position": 3,
      "sittingOut": false,
    },
    undefined,
    Object {
      "bet": 10,
      "cards": Array [],
      "chipsInPlay": 490,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player2",
      "position": 5,
      "sittingOut": false,
    },
  ],
  "seatsCount": 10,
  "smallBlind": 5,
  "toAct": 5,
}
`;

exports[`basic test with 2 players player 1 enters room 1`] = `
Object {
  "players": Object {
    "byId": Object {
      "sid_player1": Object {
        "chips": 10000,
        "id": "sid_player1",
        "name": "Player 1",
        "room": 0,
        "seat": -1,
      },
      "sid_player2": Object {
        "chips": 10000,
        "id": "sid_player2",
        "name": "Player 2",
        "seat": -1,
      },
    },
  },
  "tables": Object {
    "byId": Object {
      "0": Object {
        "activeSeatsCount": 0,
        "bigBlind": 10,
        "board": Array [],
        "gameOn": false,
        "id": 0,
        "maxActionTimeout": 60000,
        "maxBuyIn": 1000,
        "minActionTimeout": 15000,
        "minBuyIn": 500,
        "name": "Winna winna chicken dinna: 5/10 aar",
        "privateTable": false,
        "seats": Array [],
        "seatsCount": 10,
        "smallBlind": 5,
      },
    },
  },
}
`;

exports[`basic test with 2 players player 1 post small blind 1`] = `
Object {
  "activeSeatsCount": 2,
  "bigBlind": 10,
  "biggestBet": 5,
  "board": Array [],
  "button": 3,
  "deck": Array [
    "7d",
    "9c",
    "2h",
    "Ts",
    "Js",
    "3c",
    "4s",
    "8s",
    "Qs",
    "9s",
    "5c",
    "Ad",
    "9d",
    "Jc",
    "Kh",
    "5h",
    "Tc",
    "Qc",
    "Qd",
    "Ac",
    "5s",
    "9h",
    "As",
    "7c",
    "Ks",
    "3d",
    "Jh",
    "4d",
    "Qh",
    "8h",
    "6s",
    "2d",
    "6h",
    "6d",
    "6c",
    "Td",
    "7h",
    "Th",
    "8d",
    "3s",
    "3h",
    "Kc",
    "2s",
    "7s",
    "4c",
    "Jd",
    "8c",
    "2c",
    "Ah",
    "Kd",
    "5d",
    "4h",
  ],
  "gameOn": true,
  "id": 0,
  "lastPlayerToAct": 3,
  "maxActionTimeout": 60000,
  "maxBuyIn": 1000,
  "minActionTimeout": 15000,
  "minBuyIn": 500,
  "name": "Winna winna chicken dinna: 5/10 aar",
  "phase": "bigBlind",
  "privateTable": false,
  "seats": Array [
    null,
    null,
    null,
    Object {
      "bet": 5,
      "cards": Array [],
      "chipsInPlay": 495,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player1",
      "position": 3,
      "sittingOut": false,
    },
    undefined,
    Object {
      "bet": 0,
      "cards": Array [],
      "chipsInPlay": 500,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player2",
      "position": 5,
      "sittingOut": false,
    },
  ],
  "seatsCount": 10,
  "smallBlind": 5,
  "toAct": 5,
}
`;

exports[`basic test with 2 players player 1 takes seat on 3 1`] = `
Object {
  "activeSeatsCount": 1,
  "bigBlind": 10,
  "board": Array [],
  "gameOn": false,
  "id": 0,
  "maxActionTimeout": 60000,
  "maxBuyIn": 1000,
  "minActionTimeout": 15000,
  "minBuyIn": 500,
  "name": "Winna winna chicken dinna: 5/10 aar",
  "privateTable": false,
  "seats": Array [
    undefined,
    undefined,
    undefined,
    Object {
      "bet": 0,
      "cards": Array [],
      "chipsInPlay": 500,
      "inHand": false,
      "isAllIn": false,
      "playerId": "sid_player1",
      "position": 3,
      "sittingOut": false,
    },
  ],
  "seatsCount": 10,
  "smallBlind": 5,
}
`;

exports[`basic test with 2 players player 2 enters room 1`] = `
Object {
  "players": Object {
    "byId": Object {
      "sid_player1": Object {
        "chips": 10000,
        "id": "sid_player1",
        "name": "Player 1",
        "room": 0,
        "seat": -1,
      },
      "sid_player2": Object {
        "chips": 10000,
        "id": "sid_player2",
        "name": "Player 2",
        "room": 0,
        "seat": -1,
      },
    },
  },
  "tables": Object {
    "byId": Object {
      "0": Object {
        "activeSeatsCount": 0,
        "bigBlind": 10,
        "board": Array [],
        "gameOn": false,
        "id": 0,
        "maxActionTimeout": 60000,
        "maxBuyIn": 1000,
        "minActionTimeout": 15000,
        "minBuyIn": 500,
        "name": "Winna winna chicken dinna: 5/10 aar",
        "privateTable": false,
        "seats": Array [],
        "seatsCount": 10,
        "smallBlind": 5,
      },
    },
  },
}
`;

exports[`basic test with 2 players player 2 post big blind 1`] = `
Object {
  "activeSeatsCount": 2,
  "bigBlind": 10,
  "biggestBet": 10,
  "board": Array [],
  "button": 3,
  "deck": Array [
    "7d",
    "9c",
    "2h",
    "Ts",
    "Js",
    "3c",
    "4s",
    "8s",
    "Qs",
    "9s",
    "5c",
    "Ad",
    "9d",
    "Jc",
    "Kh",
    "5h",
    "Tc",
    "Qc",
    "Qd",
    "Ac",
    "5s",
    "9h",
    "As",
    "7c",
    "Ks",
    "3d",
    "Jh",
    "4d",
    "Qh",
    "8h",
    "6s",
    "2d",
    "6h",
    "6d",
    "6c",
    "Td",
    "7h",
    "Th",
    "8d",
    "3s",
    "3h",
    "Kc",
    "2s",
    "7s",
    "4c",
    "Jd",
    "8c",
    "2c",
    "Ah",
    "Kd",
    "5d",
    "4h",
  ],
  "gameOn": true,
  "id": 0,
  "lastPlayerToAct": 5,
  "maxActionTimeout": 60000,
  "maxBuyIn": 1000,
  "minActionTimeout": 15000,
  "minBuyIn": 500,
  "name": "Winna winna chicken dinna: 5/10 aar",
  "phase": "preFlop",
  "privateTable": false,
  "seats": Array [
    null,
    null,
    null,
    Object {
      "bet": 5,
      "cards": Array [],
      "chipsInPlay": 495,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player1",
      "position": 3,
      "sittingOut": false,
    },
    undefined,
    Object {
      "bet": 10,
      "cards": Array [],
      "chipsInPlay": 490,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player2",
      "position": 5,
      "sittingOut": false,
    },
  ],
  "seatsCount": 10,
  "smallBlind": 5,
  "toAct": 3,
}
`;

exports[`basic test with 2 players player 2 takes seat on 5 1`] = `
Object {
  "activeSeatsCount": 2,
  "bigBlind": 10,
  "board": Array [],
  "button": 3,
  "deck": Array [
    "7d",
    "9c",
    "2h",
    "Ts",
    "Js",
    "3c",
    "4s",
    "8s",
    "Qs",
    "9s",
    "5c",
    "Ad",
    "9d",
    "Jc",
    "Kh",
    "5h",
    "Tc",
    "Qc",
    "Qd",
    "Ac",
    "5s",
    "9h",
    "As",
    "7c",
    "Ks",
    "3d",
    "Jh",
    "4d",
    "Qh",
    "8h",
    "6s",
    "2d",
    "6h",
    "6d",
    "6c",
    "Td",
    "7h",
    "Th",
    "8d",
    "3s",
    "3h",
    "Kc",
    "2s",
    "7s",
    "4c",
    "Jd",
    "8c",
    "2c",
    "Ah",
    "Kd",
    "5d",
    "4h",
  ],
  "gameOn": true,
  "id": 0,
  "maxActionTimeout": 60000,
  "maxBuyIn": 1000,
  "minActionTimeout": 15000,
  "minBuyIn": 500,
  "name": "Winna winna chicken dinna: 5/10 aar",
  "phase": "smallBlind",
  "privateTable": false,
  "seats": Array [
    null,
    null,
    null,
    Object {
      "bet": 0,
      "cards": Array [],
      "chipsInPlay": 500,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player1",
      "position": 3,
      "sittingOut": false,
    },
    undefined,
    Object {
      "bet": 0,
      "cards": Array [],
      "chipsInPlay": 500,
      "inHand": true,
      "isAllIn": false,
      "playerId": "sid_player2",
      "position": 5,
      "sittingOut": false,
    },
  ],
  "seatsCount": 10,
  "smallBlind": 5,
  "toAct": 3,
}
`;

exports[`basic test with 2 players register Player 1 1`] = `
Object {
  "players": Object {
    "byId": Object {
      "sid_player1": Object {
        "chips": 10000,
        "id": "sid_player1",
        "name": "Player 1",
        "seat": -1,
      },
    },
  },
  "tables": Object {
    "byId": Object {
      "0": Object {
        "activeSeatsCount": 0,
        "bigBlind": 10,
        "board": Array [],
        "gameOn": false,
        "id": 0,
        "maxActionTimeout": 60000,
        "maxBuyIn": 1000,
        "minActionTimeout": 15000,
        "minBuyIn": 500,
        "name": "Winna winna chicken dinna: 5/10 aar",
        "privateTable": false,
        "seats": Array [],
        "seatsCount": 10,
        "smallBlind": 5,
      },
    },
  },
}
`;

exports[`basic test with 2 players register Player 2 1`] = `
Object {
  "players": Object {
    "byId": Object {
      "sid_player1": Object {
        "chips": 10000,
        "id": "sid_player1",
        "name": "Player 1",
        "seat": -1,
      },
      "sid_player2": Object {
        "chips": 10000,
        "id": "sid_player2",
        "name": "Player 2",
        "seat": -1,
      },
    },
  },
  "tables": Object {
    "byId": Object {
      "0": Object {
        "activeSeatsCount": 0,
        "bigBlind": 10,
        "board": Array [],
        "gameOn": false,
        "id": 0,
        "maxActionTimeout": 60000,
        "maxBuyIn": 1000,
        "minActionTimeout": 15000,
        "minBuyIn": 500,
        "name": "Winna winna chicken dinna: 5/10 aar",
        "privateTable": false,
        "seats": Array [],
        "seatsCount": 10,
        "smallBlind": 5,
      },
    },
  },
}
`;
